werkzeug/test.py

```diff
@@ -331,6 +331,8 @@ def __init__(self, path='/', base_url=None, query_string=None,
                 raise TypeError('can\'t provide input stream and data')
             if isinstance(data, text_type):
                 data = data.encode(self.charset)
+            elif isinstance(data, file):
+                data = data.read()
```

Comment:
See failures on Travis, depending on how the file was opened `read` might return a unicode or bytestring.